### PR TITLE
Bjorncs/athenz filter

### PR DESCRIPTION
**Pull request by bjorncs**
FYI @jonmv @valerijf

### CODE REVIEW

File: vespa-athenz/src/main/java/com/yahoo/vespa/athenz/zpe/AccessCheckResult.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+import com.yahoo.athenz.zpe.AuthZpeClient.AccessCheckStatus;
+
+import java.util.Arrays;
+
+/**
+ * The various types of access control results.
+ *
+ * @author bjorncs
+ */
+public enum AccessCheckResult {
```
Comment (tokle) [inline vespa-athenz/src/main/java/com/yahoo/vespa/athenz/zpe/AccessCheckResult.java]: AuthorizationResult?

File: vespa-athenz/src/main/java/com/yahoo/vespa/athenz/tls/AthenzX509CertificateUtils.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    private static final String COMMON_NAME_ROLE_DELIMITER = ":role.";
+
+    private AthenzX509CertificateUtils() {}
+
+    public static boolean isAthenzRoleCertificate(X509Certificate certificate) {
+        return isAthenzIssuedCertificate(certificate) &&
+                X509CertificateUtils.getSubjectCommonNames(certificate).get(0).contains(COMMON_NAME_ROLE_DELIMITER);
+    }
+
+    public static boolean isAthenzIssuedCertificate(X509Certificate certificate) {
```
Comment (tokle) [inline vespa-athenz/src/main/java/com/yahoo/vespa/athenz/tls/AthenzX509CertificateUtils.java]: Consider removing this check (assuming trust store is set up to only trust these issuers)

File: jdisc-security-filters/pom.xml [hunk 3]
```diff
@@ -23,6 +23,12 @@
             <version>${project.version}</version>
             <scope>provided</scope>
         </dependency>
+        <dependency>
+            <groupId>com.yahoo.vespa</groupId>
+            <artifactId>vespa-athenz</artifactId>
+            <version>${project.version}</version>
```
Comment (tokle) [inline jdisc-security-filters/pom.xml]: is project.version the correct version to use?

**Review by tokle — CHANGES_REQUESTED** [review]

File: vespa-athenz/src/main/java/com/yahoo/vespa/athenz/zpe/AccessCheckResult.java (hunk 1 above)
Comment (bjorncs) [reply vespa-athenz/src/main/java/com/yahoo/vespa/athenz/zpe/AccessCheckResult.java]: Rename done.

File: jdisc-security-filters/pom.xml (hunk 3 above)
Comment (bjorncs) [reply jdisc-security-filters/pom.xml]: Yes

**Review by tokle — APPROVED** [review]

**Pull request merged**